presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,506 @@
+    {
+        OGCGeometry geometry = getGeometry(OGCGeometry.createFromEsriGeometry(new Point(x, y), null), "ST_Point", POINT);
+        return serialize(geometry);
+    }
+
+    @Description("Returns a polygon built from Well-Known Text representation (WKT)")
+    @ScalarFunction("ST_Polygon")
+    @SqlType(NAME)
+    public static Slice stPolygon(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(OGCGeometry.fromText(input.toStringUtf8()), "ST_Polygon", POLYGON);
+        return serialize(geometry);
+    }
+
+    @Description("Returns the area of the surface if it is a polygon")
+    @ScalarFunction("ST_Area")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stArea(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_Area", POLYGON);
+        return geometry.getEsriGeometry().calculateArea2D();
+    }
+
+    @Description("Return a Geometry from Well-Known Text representation (WKT)")
+    @ScalarFunction("ST_GeometryFromText")
+    @SqlType(NAME)
+    public static Slice stGeometryFromText(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        return serialize(OGCGeometry.fromText(input.toStringUtf8()));
+    }
+
+    @SqlNullable
+    @Description("Return the Well-Known Text (WKT) representation of the geometry")
+    @ScalarFunction("ST_AsText")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice stAsText(@SqlType(NAME) Slice input)
+    {
+        return Slices.utf8Slice(deserialize(input).asText());
+    }
+
+    @Description("Returns the geometric center of a geometry")
+    @ScalarFunction("ST_Centroid")
+    @SqlType(NAME)
+    public static Slice stCentroid(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_Centroid", POLYGON);
+        SpatialReference reference = geometry.getEsriSpatialReference();
+        Envelope envelope = getEnvelope(geometry);
+        Point centroid = new Point((envelope.getXMin() + envelope.getXMax()) / 2, (envelope.getYMin() + envelope.getYMax()) / 2);
+        return serialize(OGCGeometry.createFromEsriGeometry(centroid, reference));
+    }
+
+    @Description("Return the coordinate dimension of the Geometry")
+    @ScalarFunction("ST_CoordDim")
+    @SqlType(StandardTypes.BIGINT)
+    public static long stCoordinateDimension(@SqlType(NAME) Slice input)
+    {
+        return deserialize(input).coordinateDimension();
+    }
+
+    @Description("Returns the inherent dimension of this Geometry object, which must be less than or equal to the coordinate dimension")
+    @ScalarFunction("ST_Dimension")
+    @SqlType(StandardTypes.BIGINT)
+    public static long stDimension(@SqlType(NAME) Slice input)
+    {
+        return deserialize(input).dimension();
+    }
+
+    @SqlNullable
+    @Description("Returns TRUE if the LINESTRING's start and end points are coincident")
```

Comment:
nit: for consistency sake, consider replacing LINESTRING with LineString and also mention Multi-LineString as the implementation allows both types.